net-im/swiften/swiften-4.0.ebuild

```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit scons-utils toolchain-funcs
+
+DESCRIPTION="A C++ library for implementing XMPP applications"
+HOMEPAGE="https://www.swift.im/"
+SRC_URI="https://swift.im/downloads/releases/swift-${PV}/swift-${PV}.tar.gz"
+
+LICENSE="BSD BSD-1 CC-BY-3.0 GPL-3 OFL-1.1"
+SLOT="0/4"
+KEYWORDS="~amd64 ~x86"
+IUSE="avahi ccache debug distcc doc expat experimental icu idn optimize static upnp"
+
+DEPEND="dev-db/sqlite
+	dev-libs/boost
+	dev-libs/openssl:0=
+	sys-libs/zlib
+	avahi? ( net-dns/avahi )
+	ccache? ( dev-util/ccache )
+	distcc? ( sys-devel/distcc )
+	doc? ( app-text/docbook-xsl-stylesheets
+		app-text/docbook-xml-dtd:4.5 )
+	expat? ( dev-libs/expat )
+	!expat? ( dev-libs/libxml2 )
+	icu? ( dev-libs/icu )
+	idn? ( net-dns/libidn )
+	upnp? (	net-libs/libnatpmp
+		net-libs/miniupnpc )"
+
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/swift-${PV}"
+
+src_prepare() {
+	default
+
+	# Remove those directories, as the system libs are used
+	rm -fr 3rdParty/Boost 3rdParty/DocBook 3rdParty/Expat 3rdParty/LibIDN 3rdParty/LibMiniUPnPc 3rdParty/LibNATPMP 3rdParty/Lua 3rdParty/OpenSSL 3rdParty/SCons 3rdParty/SQLite 3rdParty/Unbound 3rdParty/Zlib || die
+
+	# Remove 'Limber', 'Slimber' and 'Sluift', as compiling headless
+	rm -fr Limber Slimber Sluift || die
+}
+
+src_configure() {
+	MYSCONS=(
+		ar="$(tc-getAR)"
+		allow_warnings="$(usex debug yes no)"
+		assertions="$(usex debug yes no)"
+		build_examples="yes"
+		boost_bundled_enable="false"
+		boost_force_bundled="false"
+		cc="$(tc-getCC)"
+		ccache="$(usex ccache yes no)"
+		ccflags="${CFLAGS}"
+		coverage="$(usex debug yes no)"
+		cxx="$(tc-getCXX)"
+		cxxflags="${CXXFLAGS}"
+		debug="$(usex debug yes no)"
+		distcc="$(usex distcc yes no)"
+		$(usex doc "docbook_xml=${EPREFIX}/usr/share/sgml/docbook/xml-dtd-4.5" "" )
+		$(usex doc "docbook_xsl=${EPREFIX}/usr/share/sgml/docbook/xsl-stylesheets" "")
+		experimental="$(usex experimental yes no)"
+		experimental_ft="$(usex experimental yes no)"
+		hunspell_enable="false"
+		install_git_hooks="no"
+		libidn_bundled_enable="false"
+		libminiupnpc_force_bundled="false"
+		libnatpmp_force_bundled="false"
+		link="$(tc-getCXX)"
+		linkflags="${LDFLAGS}"
+		max_jobs="yes"
```

Comment:
What does this do?

Reply:
Pass the `max_jobs=1` to the `scons` command to speed up build time. The `max_jobs=1` argument will set the number of parallel tasks to the number of CPU cores available. Note that each parallel task increases the memory requirements for the overall build process.

Is there a better Gentoo way?

Reply:
We already specify number of jobs in `escons` (via `MAKEOPTS`/`SCONSOPTS`), so make sure it respects that.

Reply:
Looking into the code, max_jobs=1 only sets num_jobs=<max cpu count>.
But SCONSOPTS (or if empty, cleaned MAKEOPTS), sets already here -jX correctly for scons, so setting to "no" should be in that case fine. Pushed a new one.